src/tools/openapi.ts

```diff
@@ -145,16 +166,18 @@ Only use the OpenAPI tool if you need to communicate to external servers.`;
     url.pathname = join(url.pathname, path);
     // eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
     const headers: { [key: string]: string } = { Accept: "application/json" };
-    if (this.apiKey) {
-      headers["Authorization"] = `Bearer ${this.apiKey}`;
+    if (this.options.apiKey) {
+      headers["Authorization"] = `Bearer ${this.options.apiKey}`;
     }
     await this.emitter.emit("beforeFetch", { url: url });
     try {
       const response = await fetch(url.toString(), {
         body: !isEmpty(input.body) ? input.body : undefined,
         method: input.method.toLowerCase(),
         headers: headers,
-        signal: run.signal,
+        agent: !isEmpty(this.options.httpProxyUrl)
+          ? new HttpsProxyAgent(this.options.httpProxyUrl)
+          : undefined,
```

Comment:
Let user pass custom options and remove `httpProxyUrl` completely, hence remove HttpsProxyAgent dependency.